File: mcp-server-golang/registry/registry.go
package registry

import (
	"mcp-server/tools"
	"mcp-server/types"
)

var Tools []types.Tool

func init() {
	Tools = append(Tools, tools.GetTextTools()...)
	Tools = append(Tools, tools.GetDatetimeTools()...)
	Tools = append(Tools, tools.GetConverterTools()...)
	Tools = append(Tools, tools.GetRandomTools()...)
}

func FindTool(name string) *types.Tool {
	for i := range Tools {
		if Tools[i].Name == name {
			return &Tools[i]
		}
	}
	return nil
}

File: mcp-server-bunts-express/src/registry.ts
import type { Tool } from "./types/tool";
import { mathTools } from "./tools/math";
import { weatherTools } from "./tools/weather";

export const tools: Tool[] = [...mathTools, ...weatherTools];

export function findTool(name: string) {
  return tools.find((t) => t.name === name);
}
